### PR TITLE
dovecot2: Update, Bugfix, add dovecot2-sieve as subport

### PR DESCRIPTION
**Pull request by essandess**
dovecot2: Update, Bugfix, add dovecot2-sieve as subport

* Update dovecot
* Add dovecot2-sieve as subport to simplify maintenance
* Change dovecot-apns daemondo pid type to `exec`
    * exec is required because pushnotify.pl doesn't create its own pid file

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.x
Xcode 8.x

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by essandess** [comment]
This is ready to merge.

File: mail/dovecot2/Portfile [hunk 1]
```diff
@@ -79,310 +146,312 @@ configure.args      --sysconfdir=${prefix}/etc \
...
-
-        post-destroot {
-            xinstall -m 0755 -d \
-                ${destroot}/Library/LaunchDaemons
-            foreach uniquename "${daemon_uniquename} ${pushnotify_uniquename}" {
-                xinstall -m 644 -W ${filespath} ${uniquename}.plist \
-                    ${destroot}/Library/LaunchDaemons
-                reinplace "s|@PREFIX@|${prefix}|g" \
-                    ${destroot}/Library/LaunchDaemons/${uniquename}.plist
+    variant no_startupitem description {Do not install a launchd plist} {}
```
Comment (mf2k) [inline mail/dovecot2/Portfile]: We don't like to use "negative" variant names. Can you please call this variant `startupitem` instead?

Also, does this really even need to be in a variant at all? startupitems don't do anything unless the user calls them. Plenty of ports have launchd plists and don't use variants.

Comment (mf2k) [reply mail/dovecot2/Portfile]: FYI, they are normally invoked with:

`sudo port load <portname>
`

To turn it off, you use:

`sudo port unload <portname>
`

Comment (essandess) [reply mail/dovecot2/Portfile]: This variant is pre-existing and apparently old. I don’t have any insight about why it’s there and would be happy to delete it.

Comment (mf2k) [reply mail/dovecot2/Portfile]: You're right. Sorry about that. But please fix that in another PR at some point.

**Pull request merged**